src/systems/material.js

```diff
@@ -223,17 +228,32 @@ function loadImageTexture (material, src, repeat) {
      * Texture loaded. Set it.
      */
     function createTexture (texture) {
-      var repeatXY;
+      var repeatXY, offsetXY, wrapST;
       if (!(texture instanceof THREE.Texture)) { texture = new THREE.Texture(texture); }
 
+      // handle offset
+      offsetXY = offset.split(' ');
+      if (offsetXY.length === 2) {
+        texture.offset.set(offsetXY[0], offsetXY[1]);
+      }
+
       // Handle UV repeat.
       repeatXY = repeat.split(' ');
       if (repeatXY.length === 2) {
-        texture.wrapS = THREE.RepeatWrapping;
-        texture.wrapT = THREE.RepeatWrapping;
+        if (!wrap) {

```

Comment:
if you declare this variable in the schema yo always have one defined and this if won't be necessary since THREE.RepeatWrapping will be the default